php/libraries/User.class.inc

```diff
@@ -41,15 +41,18 @@ class User extends UserPermissions
      * @return User
      * @access public
      */
-    static function &factory($username)
+    static function &factory($username) : User
     {
         $obj = new User;
 
         // right off, set the username
         $obj->userInfo['UserID'] = $username;
 
         // get the user's permissions
-        $obj->select($username);
+        if ($obj->select($username) === false) {
```

Comment:
@driusan this breaks the survey module which is relying on permissions to be loaded in order to load the feedback panel...